### PR TITLE
Refactoring cmd_devfile_watch_tests

### LINKED ISSUE #5161: Refactor test-cmd-devfile-watch
/kind tests

## Acceptance Criteria 
- [ ] test-cmd-devfile-watch should use new test approach and run successfully.

### PR DESCRIPTION
**Pull request by rnapoles-rh**
**What type of PR is this?**

> /kind tests

**What does this PR do / why we need it**:
Refactoring cmd_devfile_watch_tests tests 

**Which issue(s) this PR fixes**:

Fixes #5161 

**PR acceptance criteria**:

- [x] Unit test 

- [x] Integration test 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by feloy** [comment]
/test psi-kubernetes-integration-e2e

```
[ssh:Fedora-32-minikube] [Fail] odo service command tests for OperatorHub Operators are installed in the cluster a specific operator is installed when a nodejs component is created when a Redis instance is created with no name and inlined flag is used when odo push is executed [BeforeEach] when a link is created with the service Should fail to link it again 
[ssh:Fedora-32-minikube] /home/fedora/amqp_ci_rcv_odo-minikube-pr-build_PR_5219/KdBkf/repo/tests/helper/helper_cmd_wrapper.go:100
[ssh:Fedora-32-minikube] 
[ssh:Fedora-32-minikube] [Fail] odo link command tests for OperatorHub Operators are installed in the cluster when a component and a service are deployed when a link between the component and the service is created when odo push is executed [BeforeEach] should find files in component container 
```

File: tests/integration/devfile/cmd_devfile_watch_test.go [hunk 1]
```diff
@@ -31,105 +31,61 @@ var _ = Describe("odo devfile watch command tests", func() {
...
+	When("running help for watch command", func() {
 		It("should display the help", func() {
 			appHelp := helper.Cmd("odo", "watch", "-h").ShouldPass().Out()
 			helper.MatchAllInOutput(appHelp, []string{"Watch for changes", "git components"})
 		})
 	})
 
-	Context("when executing watch without pushing a devfile component", func() {
-		It("should fail", func() {
+	When("executing watch without pushing a devfile component", func() {
```
Comment (feloy) [inline tests/integration/devfile/cmd_devfile_watch_test.go:41]: The `When` contents should explain what is doing the `BeforeEach` part only, here something similar to `When("a component is created"`

File: tests/integration/devfile/cmd_devfile_watch_test.go [hunk 2]
```diff
@@ -31,105 +31,61 @@ var _ = Describe("odo devfile watch command tests", func() {
...
+		BeforeEach(func() {
 			helper.Chdir(commonVar.OriginalWorkingDirectory)
 			helper.Cmd("odo", "create", "--project", commonVar.Project, "--context", commonVar.Context, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-registry.yaml")).ShouldPass()
+		})
+		It("should fail", func() {
```
Comment (feloy) [inline tests/integration/devfile/cmd_devfile_watch_test.go:46]: `should fail when odo push is not executed`?

File: tests/integration/devfile/cmd_devfile_watch_test.go [hunk 3]
```diff
@@ -31,105 +31,61 @@ var _ = Describe("odo devfile watch command tests", func() {
...
 		})
-
 		It("should error out on devfile flag", func() {
 			helper.Cmd("odo", "watch", "--devfile", "invalid.yaml").ShouldFail()
 		})
 	})
 
-	Context("when executing odo watch after odo push", func() {
-		It("should listen for file changes", func() {
+	When("executing odo watch", func() {
```
Comment (feloy) [inline tests/integration/devfile/cmd_devfile_watch_test.go:55]: When("a component is created and pushed"

**Review by feloy — CHANGES_REQUESTED** [review]

**Comment by rnapoles-rh** [comment]
Closing as it is replaced with https://github.com/openshift/odo/pull/5244

**Pull request closed without merging**